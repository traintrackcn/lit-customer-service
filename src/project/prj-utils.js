export const getPlans = (prj) => {
    const description = prj.get('description');
    const tag = 'plan';
    const content = getContentBetweenTag(description, tag);
    console.log('content ->', content);
    if (content){
        return content.match(/\w+\s.+/g);
    }

    return [];
}

export const getConfig = (prj, key) => {
    const description = prj.get('description');
    const tag = 'config';
    const content = getContentBetweenTag(description, tag);

    if (content){
        const regex = new RegExp(key+'-->.+', 'g');
        console.log('regex ->', regex);
        // return content.match(/code+-->\w+/g);
        let matches = content.match(regex);
        console.log('matches ->', JSON.stringify(matches, null, 2));

        if (matches && matches.length > 0){
            let value = matches[0].replace(new RegExp(key+'-->', 'g'), '');
            return value;
        }

        return;
    }

    return;
}

export const getContentBetweenTag = (str, tag) => {
    // const description = prj.get('description');
    const regex = new RegExp('\\['+tag+'\\]([\\s\\S]*?)\\['+tag+'\\]', 'g');
    // const regex = /\[plan\]([sS]*?)\[plan\]/g;
    console.log('regex -> ', regex);
    const matches =  str.match(regex);
    console.log('matches -> '+JSON.stringify(matches, null, 2));
    if (matches && matches.length > 0){
        let match = matches[0];
        return match;
    }
    return;
}