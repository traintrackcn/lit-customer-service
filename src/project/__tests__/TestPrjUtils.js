import { getPlans, getCode, getConfig } from '../prj-utils';
import { fromJS } from 'immutable';

describe('', () => {

    it('', ()=>{
        let description = `

        **config-begin**
        platform-->ios
        customer-->Customer A|Customer B
        **config-end**

        [config]
        code-->MON
        customer-->Customer A|Customer B
        [config]
        
        [plan]
        Elite 2018.9~2018.12
        Basic 2017.8~2017.8
        Basic 2015.11~2017.1
        [plan]
        
        
        `;

        var prj = {
            "description": description
        };

        // const regex = /\[plan\-begin\]\w+\[plan\-end\]/g;
        // const match = getPlans()
        const plan = getPlans(fromJS(prj), 'plan');
        expect(plan).toMatchSnapshot();

        const code = getConfig(fromJS(prj), 'code');
        expect(code).toMatchSnapshot();

        const customer = getConfig(fromJS(prj), 'customer');
        expect(customer).toMatchSnapshot();

    });
    

});