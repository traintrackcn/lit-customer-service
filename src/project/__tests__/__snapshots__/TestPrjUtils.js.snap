// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  "Elite 2018.9~2018.12",
  "Basic 2017.8~2017.8",
  "Basic 2015.11~2017.1",
]
`;

exports[` 2`] = `"MON"`;

exports[` 3`] = `"Customer A|Customer B"`;
